lib/improver/utilities/save.py

```diff
@@ -43,8 +43,35 @@ def append_metadata_cube(cubelist):
     Returns:
         iris.cube.Cubelist with appended metadata cube
     """
+
+    keys_for_global_attr = {}
+
+    # Set up a basic prefix cube
     prefix_cube = iris.cube.Cube(0, long_name='prefixes',
                                  var_name='prefix_list')
+
+    # Iterate through the cubelist cubes attributes (using dictionary
+    # comprehension), collecting attributes that we want to be global
+    # attributes in a resulting netCDF file.
+    for cube in cubelist:
+        keys = cube.attributes
+        keys_for_global_attr = {k: v for k, v in keys.items()
+                                if k is 'um_version'
+                                or k is 'institution'
+                                or k is 'source'
+                                or k is 'grid_id'
+                                or k is 'history'
+                                or k is 'Conventions'
+                                or k is 'title'}
```

Comment:
OK so I've just tried this out using a list of global keys in append_metadata_cube (which could or course be passed in as argument ultimately). 

Your code seems to be producing a set rather than dictionary which caused problems when adding the keys to the prefix cube, so I added a line to produce a dictionary from the set which also set's the global key's value to ' '. I'm not sure if this is what you meant? 

For the attributes to kept as global I'm not sure if the keys and the values of the attributes have to be the same or just the keys? (Its not clear to me from the Iris2 docs....)

I'll push up this commit so you can see what you think.......